packages/@orbit/core/src/task-queue.ts

```diff
@@ -212,12 +212,17 @@ export default class TaskQueue implements Evented {
    * If `autoProcess` is enabled, this will automatically trigger processing of
    * the queue.
    */
-  skip(): Promise<void> {
+  skip(e?: Error): Promise<void> {
     return this._reified
       .then(() => {
         this._cancel();
         this._tasks.shift();
-        this._processors.shift();
+        let processor = this._processors.shift();
```

Comment:
I just found a situation where `this._processors` was an empty array and therefore the `!processor.settled` in the next line failed. 
Sadly I'm not sure what did I do to end up in that state and now I can't reproduce it, but I thought it was worth mentioning in case we should add a guard for `processor` being undefined.

Reply:
@cibernox Ah, that's a good catch. We should guard against that scenario, since I think that `skip` should still succeed in that case. Please open an issue (or PR) so we don't lose track of this.

Reply:
Just opened #686 for that.